framework/Shell/Actions/TFlushCachesAction.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * TFlushCachesAction class file
+ *
+ * @author Brad Anderson <belisoful@icloud.com>
+ * @link https://github.com/pradosoft/prado
+ * @license https://github.com/pradosoft/prado/blob/master/LICENSE
+ * @package Prado\Shell\Actions
+ */
+
+namespace Prado\Shell\Actions;
+
+use Prado\Caching\ICache;
+use Prado\Shell\TShellAction;
+
+/**
+ * This command clears all application modules implementing ICache.
+ *
+ * @author Brad Anderson <belisoful[at]icloud[dot]com>
+ * @package Prado\Shell\Actions
+ * @since 4.2.0
+ */
+class TFlushCachesAction extends TShellAction
+{
+	protected $action = 'flushcaches';
+	protected $parameters = [];
+	protected $optional = ['directory'];
+	protected $description = 'Flushes all application TCache modules. Use case: upgrading a performance mode website by clearing out the old cache.';
+
+	/**
+	 * @param array $args parameters
+	 * @return bool
+	 */
+	public function performAction($args)
+	{
+		$app = null;
+		if (count($args) > 1) {
+			if (false === ($xml = $this->getAppConfigFile($args[1]))) {
+				return false;
+			}
+			$app = $this->initializePradoApplication($args[1]);
+		}
+		$app->onLoadState();
```

Comment:
```
PHP Fatal error:  Uncaught Error: Call to a member function onLoadState() on null in /var/www/htdocs/prado-master/framework/Shell/Actions/TFlushCachesAction.php:43

```
The directory parameter should be mandatory. 
Still, better check if $app is null before using it.

Reply:
Agreed.  It was really weak sauce to begin with, I didn't both proofing it, obviously.  I didn't want to spend eternity on this refactor even though it's looking like I might have to spend some time making it reasonable.  lol.

Reply:
updated